### PR TITLE
feat: try to avoid string copy in insert-values again

### PR DESCRIPTION
**Pull request by ygf11**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

## Changelog

- Improvement
- Performance Improvement

## Related Issues

Related #4413

### CODE REVIEW

File: common/io/src/buffer/checkpoint_reader.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@ use super::checkpoint_read::CheckpointRead;
 use super::BufferRead;
 
 // This is dynamic checkpoint buffer reader
-pub type CpBufferReader = CheckpointReader<Box<dyn BufferRead>>;
+pub type CpBufferReader<'a> = CheckpointReader<Box<dyn BufferRead + 'a>>;
```
Comment (sundy-li) [inline common/io/src/buffer/checkpoint_reader.rs:22]: Impressive！

**Comment by sundy-li** [comment]
```
--- /workspace/tests/suites/0_stateless/02_function/02_0012_function_datetimes.result	2022-04-07 04:21:38.1509[229](https://github.com/datafuselabs/databend/runs/5862463614?check_suite_focus=true#step:3:229)37 +0000
+++ /workspace/tests/suites/0_stateless/02_function/02_0012_function_datetimes.stdout	2022-04-07 04:23:08.796789523 +0000
@@ -1,3 +1,4 @@
+ERROR 1105 (HY000) at line 196: Code: 1005, displayText = sql parser error: Expected (, found: now.
 1
 1
 2021-08-30 10:47:42	1
@@ -160,6 +161,5 @@
 1
 1
 1
-1	1	1	1
 2022-04-02 15:10:28	2022-04-02 15:10:28.000
 2022-04-02 15:10:28	2022-04-02 15:10:28.221
```

**Comment by ygf11** [comment]
@sundy-li It is in the sqlparser. I will fix it.

**Comment by sundy-li** [comment]
Performance improved by around 50%.

**Comment by sundy-li** [comment]
@mergify update

**Review by sundy-li — APPROVED** [review]

**Comment by Xuanwo** [comment]
@mergify update

**Review by BohuTANG — APPROVED** [review]
🚀

**Comment by Xuanwo** [comment]
Oh, my fault. mergify not works as expected: please wait for https://github.com/datafuselabs/databend/pull/4752

**Pull request merged**